Doc/c-api/code.rst

```diff
@@ -51,3 +51,11 @@ bound into a function.
    Return a new empty code object with the specified filename,
    function name, and first line number.  It is illegal to
    :func:`exec` or :func:`eval` the resulting code object.
+
+.. c:function:: int PyCode_Addr2Line(PyCodeObject *co, int byte_offset)
+
+    Return the line number of the of instruction that occurs on or before ``byte_offset`` and ends after it.
+    This function is kept for backwards compatibility, its use is not recommended.
```

Comment:
If this function should not be used, I suggest to officially deprecate it. Add ".. deprecated:: 3.10", add it to https://docs.python.org/dev/whatsnew/3.10.html#id3 and add Py_DEPRECATED() in the header file where it's defined. Usually, we wait at two Python releases before removing a function (PEP 387).

I checked Tools/gdb/libpython.py: it reimplements this function :-)

```
    def addr2line(self, addrq):
        '''
        Get the line number for a given bytecode offset

        Analogous to PyCode_Addr2Line; translated from pseudocode in
        Objects/lnotab_notes.txt
        '''
```